app/src/pages/inside/projectSettingsPageContainer/projectSettingsPageContainer.jsx

```diff
@@ -14,39 +14,161 @@
+        },
+      }),
+      {},
     );
-  }
-}
+  };
+
+  const createItemsConfig = () => {
+    const extensionTabs = createExtensionTabs();
+    const tabsConfig = {
+      [GENERAL]: {
+        name: formatMessage(messages.general),
+        link: createTabLink(GENERAL),
+        component: <GeneralTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.GENERAL_TAB,
+        mobileDisabled: true,
+      },
+      [INTEGRATIONS]: {
+        name: formatMessage(messages.integrations),
+        link: createTabLink(INTEGRATIONS),
+        component: <IntegrationsTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.INTEGRATIONS_TAB,
+      },
+      [NOTIFICATIONS]: {
+        name: formatMessage(messages.notifications),
+        link: createTabLink(NOTIFICATIONS),
+        component: <NotificationsTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.NOTIFICATIONS_TAB,
+        mobileDisabled: true,
+      },
+      [DEFECT]: {
+        name: formatMessage(messages.defect),
+        link: createTabLink(DEFECT),
+        component: <DefectTypesTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.DEFECT_TYPE_TAB,
+        mobileDisabled: true,
+      },
+      [ANALYSIS]: {
+        name: formatMessage(messages.analysis),
+        link: createTabLink(ANALYSIS),
+        component: <AutoAnalysisTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.AUTO_ANALYSIS_TAB,
+        mobileDisabled: true,
+      },
+      [PATTERN_ANALYSIS]: {
+        name: formatMessage(messages.patternAnalysis),
+        link: createTabLink(PATTERN_ANALYSIS),
+        component: <PatternAnalysisTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.PATTERN_ANALYSIS_TAB,
+        mobileDisabled: true,
+      },
+      [DEMO_DATA]: {
+        name: formatMessage(messages.demoData),
+        link: createTabLink(DEMO_DATA),
+        component: <DemoDataTab />,
+        eventInfo: SETTINGS_PAGE_EVENTS.DEMO_DATA_TAB,
+        mobileDisabled: true,
+      },
+    };
+    if (!canSeeDemoData(accountRole, userRole)) {
+      delete tabsConfig[DEMO_DATA];
+    }
+    Object.keys(extensionTabs).forEach((tab) => {
+      if (tabsConfig[tab]) {
+        tabsConfig[tab].component = extensionTabs[tab].component;
+
+        delete extensionTabs[tab];
+      }
+    });
+    return { ...tabsConfig, ...extensionTabs };
+  };
+
+  const navigation = useMemo(() => {
+    const itemsConfig = createItemsConfig();
+    const title = <FormattedMessage id="SettingsPage.title" defaultMessage="Project Settings" />;
+    return <Navigation items={itemsConfig} title={title} />;
+  }, [createItemsConfig]);
+
+  const content = useMemo(() => {
+    const itemsConfig = createItemsConfig();
+    if (!activeTab || !itemsConfig[activeTab]) {
+      const firstItemName = Object.keys(itemsConfig)[0];
+      dispatch(itemsConfig[firstItemName].link);
+      return null;
+    }
+    return itemsConfig[activeTab].component;
+  }, [activeTab, createItemsConfig, dispatch]);
```

Comment:
createItemsConfig in the dependencies. And createItemsConfig will be called every time activeTab  is changed.
Maybe make createItemsConfig useCallback?